network-grpc/src/client.rs

```diff
@@ -335,7 +347,26 @@ where
     }
 }
 
-impl<T, S, E> BlockService for Client<T, S, E>
+impl<T> FromResponse<gen::node::GossipMessage> for (NodeId, T)
+where
+    T: Gossip,
+{
+    fn from_response(res: gen::node::GossipMessage) -> Result<(NodeId, T), core_client::Error> {
+        let node_id = match res.node_id {
+            None => panic!("no node type"),
+            Some(gen::node::gossip_message::NodeId { content }) => {
+                match NodeId::from_slice(&content) {
+                    Ok(node_id) => node_id,
+                    Err(_v) => panic!("incorrect node encoding"),
```

Comment:
We shall not panic from invalid network requests.
Respond with error carrying `Code::InvalidArgument` (to be added in network-core if not already).